jib-core/src/test/java/com/google/cloud/tools/jib/builder/steps/BuildManifestListOrSingleManifestStepTest.java

```diff
@@ -0,0 +1,157 @@
+/*
+ * Copyright 2020 Google LLC.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not
+ * use this file except in compliance with the License. You may obtain a copy of
+ * the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+package com.google.cloud.tools.jib.builder.steps;
+
+import com.google.cloud.tools.jib.blob.BlobDescriptor;
+import com.google.cloud.tools.jib.builder.ProgressEventDispatcher;
+import com.google.cloud.tools.jib.configuration.BuildContext;
+import com.google.cloud.tools.jib.event.EventHandlers;
+import com.google.cloud.tools.jib.image.Image;
+import com.google.cloud.tools.jib.image.Layer;
+import com.google.cloud.tools.jib.image.json.ManifestTemplate;
+import com.google.cloud.tools.jib.image.json.V22ManifestListTemplate;
+import com.google.cloud.tools.jib.image.json.V22ManifestTemplate;
+import java.io.IOException;
+import java.util.Arrays;
+import org.junit.Assert;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.Mockito;
+import org.mockito.junit.MockitoJUnitRunner;
+
+/** Tests for {@link BuildManifestListOrSingleManifest}. */
+@RunWith(MockitoJUnitRunner.class)
+public class BuildManifestListOrSingleManifestStepTest {
+
+  @Mock private ProgressEventDispatcher.Factory progressDispatcherFactory;
+  @Mock private BuildContext buildContext;
+  @Mock private Layer layer;
+
+  private Image image1;
+  private Image image2;
+
+  @Before
+  public void setUp() {
+    Mockito.when(buildContext.getEventHandlers()).thenReturn(EventHandlers.NONE);
+    Mockito.doReturn(V22ManifestTemplate.class).when(buildContext).getTargetFormat();
+    Mockito.when(layer.getBlobDescriptor()).thenReturn(new BlobDescriptor(0, null));
+
+    image1 =
+        Image.builder(V22ManifestTemplate.class)
+            .setArchitecture("amd64")
+            .setOs("linux")
+            .addLayer(new PreparedLayer.Builder(layer).setName("resources").build())
+            .build();
+    image2 =
+        Image.builder(V22ManifestTemplate.class)
+            .setArchitecture("arm64")
+            .setOs("ubuntu")
```

Comment:
nit: let's say `"windows"` to eliminate any potential misunderstanding. (Ubuntu is linux).